### ISSUE TITLE
Hapi V3.3.0 is not JDK 7 compatible

### ISSUE THREAD

**Issue by Jmacek** [jamesagnew/hapi-fhir#1107]
NOTE: Before filing a ticket, please see the following URL:
https://github.com/jamesagnew/hapi-fhir/wiki/Getting-Help

**Describe the bug**
The changelogs for Hapi v3.3.0 mention that it is JDK 7 compatible, but it requires a version of Guava that is past JDK 7 support (Guava 23.0). You can't simply exclude this if you're using OSGI like I am because that version is still trying to be pulled in through the manifest. Is there a way we can exclude this from the manfiest classpath or downgrade to 21.0 which is the last supported version of Guava to support JDK 7?

**To Reproduce**
1. Try to run the application on JRE 7.
2. Get major.minor version error 52.0 on Guava

**Expected behavior**
Should work on JRE/JDK 7

**Environment (please complete the following information):**
 - HAPI FHIR Version: 3.3.0
 - OS: Windows 10

**Additional context**
If there were a solution to excluding the import-package from the manifest I would take that too, but as it stands 3.3.0 is not technically JDK 7 compatible.

**Comment by jamesagnew** [comment]
Ah this is interesting. Would you be willing to submit a pull request against the v3.3.0 tag to correct this? We could release a 3.3.1 update to correct it.

**Comment by jamesagnew** [comment]
Ah this is interesting. Would you be willing to submit a pull request against the v3.3.0 tag to correct this? We could release a 3.3.1 update to correct it.

**Comment by Jmacek** [comment]
I can do that, I should note that I was able to get it working on JDK 7 by excluding guava and including the guava 23.0-android variant (which is the fork of guava that still retains support for jdk 7 after guava dropped support for it) which satisfies the requirement on guava 23.0 in the manifest. 

However I don’t think it’s an ideal situation to have to include your own variant of a library to get something working on a jdk version that is supposed to be supported. Whether we downgrade to guava 21.0 or modify the version to use 23.0-android by default I’ll leave up to you to decide, but yes I’m fine with making a PR off of the 3.3.0 tag once we decide which route to take.

**Comment by restevez-chs** [comment]
@Jmacek hi. i have been reading this issue and it seems one of the candidate solution is workable for you: use 23.0-android for guava dependency

i saw that you last mentioned that you could submit a PR. 

are you able to submit a PR still?

if not, i have some spare cycles and willing to make the proposed change, submit the PR and see what happens.

let me know. thanks!

**Comment by restevez-chs** [comment]
i have a PR branch ready to go but trying to figure out how to create a PR against the v3.3.0 tag. is there a trick to it?

https://github.com/restevez-chs/hapi-fhir/tree/issue-1107-v3.3.1-release-candidate

at the moment i'm running a `mvn clean install` locally

**Comment by Jmacek** [comment]
@restevez-chs hey thanks sorry I’ve been swamped lately and haven’t been able to get around to making the patchback. 

As for making a PR against a tag you can’t do that so I assume that what @jamesagnew was looking for was for whoever made the change to make a branch off of the 3.3.0 tag and make a PR into that from your feature branch.

Go ahead and add me as a reviewer and I’ll be glad to look over it but I’m pretty sure that’s what James wanted.

**Comment by Jmacek** [comment]
Yeah you’re right someone with permission to the original repo would need to make a branch off of 3.3.0. I forgot that was the whole reason I hadn’t done this yet lol...well we’ll have to wait for James then.

**Comment by restevez-chs** [comment]
yup. we'll see how it goes!

**Comment by restevez-chs** [comment]
@Jmacek i am not able to add reviewers. i have had the PRs open for a while now but have not heard anything.

**Comment by Jmacek** [comment]
I see, well unfortunately sometimes that's what happens on large open source projects. Maybe we can track down another collaborator to get this moved forward?

**Comment by areklalo** [comment]
Hi @fil512,
could you take a look on this?

**Comment by areklalo** [comment]
@jamesagnew @fil512 @patrick-werner 
What can we do something to move forward with it?
